File: january2025/january13.java
class Solution {
    public int maxWater(int[] arr) {
        int l = 0;
        int r = arr.length - 1;
        int res = 0;

        while (l < r) {
            res = Math.max(res, (r - l) * Math.min(arr[l], arr[r]));
            if (arr[l] < arr[r]) {
                l++;
            } else {
                r--;
            }
        }

        return res;
    }
}

File: january2025/january13.py

class Solution:
    def maxWater(self, arr):
        # code here
        l,r,res=0,len(arr)-1,0
        while l<r:
            res=max(res,(r-l)*min(arr[l],arr[r])
            l,r=(l+1,r) if arr[l]<arr[r] else (l,r-1)
        return res


#{ 
 # Driver Code Starts
#Initial template for Python 3

import math


def main():
    t = int(input())
    while (t > 0):

        arr = [int(x) for x in input().strip().split()]
        obj = Solution()
        print(obj.maxWater(arr))

        t -= 1
        print("~")


if __name__ == "__main__":
    main()

# } Driver Code Ends
